src/Constraint/MultiConstraint.php

```diff
@@ -132,6 +132,34 @@ public function matches(ConstraintInterface $provider)
         return true;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function isSubsetOf(ConstraintInterface $constraint)
+    {
+        if ($constraint instanceof EmptyConstraint) {
+            return true;
+        }
+
+        if (true === $this->conjunctive) {
+            foreach ($this->constraints as $c) {
+                if (!$c->isSubsetOf($constraint)) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        foreach ($this->getConstraints() as $c) {
+            if ($c->isSubsetOf($constraint)) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
this looks incomplete to me, in case the MultiConstraint is not optimized to merge ranges.